### PR TITLE
Add misc type annotations and Python 3.10+ updates

### PR DESCRIPTION
**Pull request by mborgerson**

### CODE REVIEW

File: angr/knowledge_plugins/cfg/cfg_node.py [hunk 1]
```diff
@@ -87,13 +90,13 @@ def __init__(
         __repr__.
         """
 
-        self.addr = addr
+        self.addr: int = addr
```
Comment (twizmwazin) [inline angr/knowledge_plugins/cfg/cfg_node.py:93]: Java uses non-int addrs

Comment (ltfish) [reply angr/knowledge_plugins/cfg/cfg_node.py:93]: I think we need an `AddressType = int | SootAddressDescriptor`.

Comment (twizmwazin) [reply angr/knowledge_plugins/cfg/cfg_node.py:93]: That or make cfg-related types generic over an address type

Comment (mborgerson) [reply angr/knowledge_plugins/cfg/cfg_node.py:93]: Let's go with @ltfish 's suggestion for now and improve address abstraction later

**Pull request merged**